vdjtools

```diff
@@ -2,11 +2,24 @@
 
```

Comment:
Overall, the script is well written and follows good practices. However, there are a few areas that could be improved for better readability and maintainability.

1. **Use of Comments**: The script could benefit from more comments explaining what each section of the script does. This would make it easier for others to understand and maintain.

2. **Variable Naming**: Some variables could have more descriptive names. For example, `jVersion1` and `jVersion2` could be renamed to `javaMajorVersion` and `javaMinorVersion` respectively.

3. **Error Handling**: The script could have better error handling. For example, if the `java` command is not found, the script should exit with an appropriate error message.

4. **Use of Functions**: The script has a function `readlinkUniversal` defined but it's only used once. If it's not going to be reused, the code could just be included in the main script. If it is going to be reused, it might be better placed in a separate script that can be sourced.

5. **Use of Magic Numbers**: The script has a few "magic numbers", like `100` and `12000`. It would be better to define these as named constants at the top of the script, to make it clearer what they represent and easier to change if needed.

6. **Consistent Formatting**: The script could benefit from consistent formatting. For example, sometimes there are spaces around `=` in assignments, and sometimes there are not. Choose one style and stick to it.

Here's how these suggestions could be incorporated:

```bash
#!/bin/bash

# Constants
JAVA="java"
DELTA=100
TARGET_XMX=12000

# Extract Java version
JAVA_VERSION=$($JAVA -version 2>&1 | grep version | awk '{ print $3 }' | sed 's:"::g')
JAVA_MAJOR_VERSION=$(echo "$JAVA_VERSION" | sed "s/\s*\([^.]*\)\..*/\1/")
JAVA_MINOR_VERSION=$(echo "$JAVA_VERSION" | sed "s/.*1\.\(.*\)\..*/\1/")

# Check Java version
if [[ JAVA_MAJOR_VERSION -eq 1 ]]; then
    if [[ $JAVA_MINOR_VERSION -lt 7 ]]; then
        echo "Wrong version of Java. Please use Java 7 or higher."
        exit 1
    fi

    if [[ $JAVA_MINOR_VERSION -lt 8 ]]; then
        echo "WARNING: Several issues have been detected while using Java 7. Please consider updating to Java 8 or higher."
    fi
elif [[ JAVA_MAJOR_VERSION -lt 9 ]]; then
    echo "Wrong version of Java. Please use Java 7 or higher."
    exit 1
fi

# Rest of the script...
```

This makes the script more readable and maintainable.